src/main/java/com/capstone/service/impl/RegistrationServiceImpl.java

```diff
@@ -235,4 +246,53 @@ public ApiResponseDto<String> resendInvitation(String email) {
         }
     }
 
+    private void assignSpecializationsToMentor(User mentor, List<UUID> specializationIds) {
+        for (UUID specializationId : specializationIds) {
+            // Use findBySpecId instead of findById
+            Specialization specialization = specializationRepository.findBySpecId(specializationId)
+                    .orElseThrow(() -> new SpecializationNotFoundException("Specialization not found: " + specializationId));
+
+            // Check if relationship already exists using specId
+            if (!mentorSpecializationRepository.existsByUserIdAndSpecializationSpecId(mentor.getId(), specializationId)) {
+                // Create MentorSpecialization relationship
+                MentorSpecialization mentorSpec = MentorSpecialization.builder()
+                        .user(mentor)
+                        .specialization(specialization)
+                        .build();
+
+                mentorSpecializationRepository.save(mentorSpec);
+                log.debug("Assigned specialization '{}' to mentor '{}'", specialization.getSpecName(), mentor.getEmail());
+            }
+        }
```

Comment:
This is an expensive operation. You could use `upsert` instead and let the DB choose which to insert and which to ignore, given you have an enforced relationship in the join table like `UNIQUE (userId, specializationId)`.

Example of upserting:
```Java
@Modifying
@Query(value = """
    INSERT INTO mentor_specialization (user_id, specialization_id)
    VALUES (:userId, UNNEST(:specIds))
    ON CONFLICT (user_id, specialization_id) DO NOTHING
    """, nativeQuery = true)
void upsertMentorSpecializations(@Param("userId") UUID userId, @Param("specIds") UUID[] specIds);

```